host-controller/src/main/java/org/jboss/as/host/controller/discovery/S3Discovery.java

```diff
@@ -117,21 +111,7 @@ public void allowDiscovery(List<DomainControllerManagementInterface> interfaces)
             for (DomainControllerManagementInterface managementInterface : interfaces) {
                 data.add(new DomainControllerData(managementInterface.getProtocol().toString(), managementInterface.getHost(), managementInterface.getPort()));
             }
-            Collections.sort(data, new Comparator<DomainControllerData>() {
-                @Override
-                public int compare(DomainControllerData data, DomainControllerData otherData) {
-                    if (Protocol.REMOTE.toString().equals(data.getProtocol())) {
-                        return 1;
-                    }
-                    if (Protocol.HTTPS_REMOTING.toString().equals(data.getProtocol())) {
-                        if (Protocol.REMOTE.toString().equals(otherData.getProtocol())) {
-                            return -1;
-                        }
-                        return 1;
-                    }
-                    return -1;
-                }});
```

Comment:
Why is this sorting being removed? This is what ensures that legacy hosts see the remote protocol that they understand.